magicai/Prompts.php

```diff
@@ -0,0 +1,295 @@
+<?php
+/**
+ * Helper class for creating the AI prompts for text generation.
+ *
+ * @package Tutor\MagicAI
+ * @author Themeum <support@themeum.com>
+ * @link https://themeum.com
+ * @since 3.0.0
+ */
+
+namespace Tutor\MagicAI;
+
+use TUTOR\Input;
+use Tutor\MagicAI\Constants\Models;
+
+/**
+ * Helper class for generating AI prompts for text generation
+ *
+ * @since 3.0.0
+ */
+final class Prompts {
+	/**
+	 * Create the system message for generating text content using tone, format, language, etc.
+	 *
+	 * @return string
+	 * @since   3.0.0
+	 */
+	private static function create_system_message() {
+		$input = array(
+			'tone'       => Input::post( 'tone', 'formal' ),
+			'format'     => Input::post( 'format', 'essay' ),
+			'language'   => Input::post( 'language', 'english' ),
+			'characters' => Input::post( 'characters', 250 ),
+			'is_html'    => Input::post( 'is_html', false, Input::TYPE_BOOL ),
+		);
+
+		$system_content = 'You are a friendly and helpful assistant. You will be provided a prompt, and your task to generate a {format}. The content will be in the {language} language and has a {tone} tone. Make sure the content will not exceed the length of {characters} characters.';
+
+		if ( ! $input['is_html'] ) {
+			$system_content .= ' Please respond with plain text only. Do not use markdown or HTML.';
+		}
+
+		foreach ( $input as $key => $value ) {
+			$system_content = str_replace( "{$key}", $value, $system_content );
+		}
+
+		return $system_content;
+	}
+
+	/**
+	 * Prepare the input array for generating text content from the request prompt.
+	 *
+	 * @return array
+	 * @since 3.0.0
+	 */
+	public static function prepare_text_generation_input() {
+		$prompt = Input::post( 'prompt', '' );
+
+		return array(
+			'model'       => Models::GPT_4O,
+			'messages'    => array(
+				array(
+					'role'    => 'system',
+					'content' => self::create_system_message(),
+				),
+				array(
+					'role'    => 'user',
+					'content' => $prompt,
+				),
+			),
+			'temperature' => 0.7,
+		);
+	}
+
+	/**
+	 * Prepare the input array for translating content to a specific language.
+	 *
+	 * @return array
+	 * @since 3.0.0
+	 */
+	public static function prepare_translation_input() {
+		$content        = Input::post( 'content', '' );
```

Comment:
`Reusability`
We can make the method more loosely coupled by passing parameters directly, rather than depending on the request's POST data.